### PR TITLE
Adding a KerasMLPRegression class to go with the KerasMLPClassifier.

### LINKED ISSUE #1706: [TASK] Add Keras Regression for non-timeseries data
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
There should be a Keras Regression for non-timeseries data

**Describe the solution you'd like**
Add a KerasMLPRegression similar to the KerasMLPClassifier

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
This allows regression to be done for non-timeseries data.

--------
Pull Request Description
--------
##### What issue does this change request address? 
Closes #1706 

##### What are the significant changes in functionality due to this change request?
Adds a Keras MLP Regression class, and fixes a bug in the Keras Regression.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: framework/SupervisedLearning/KerasRegression.py [hunk 1]
```diff
@@ -145,7 +151,6 @@ def evaluate(self,edict):
           featureValuesShape = fval.shape
         if featureValuesShape != fval.shape:
           self.raiseAnError(IOError,'In training set, the number of values provided for feature '+feat+' are not consistent to other features!')
-        self._localNormalizeData(values,names,feat)
```
Comment (joshua-cogliati-inl) [inline framework/SupervisedLearning/KerasRegression.py:148]: This was a bug since it changed the normalization constants based on the evaluation data.

**Review by aalfonsi — APPROVED** [review]

**Review by wangcj05 — APPROVED** [review]
PR is good.

**Comment by wangcj05** [comment]
Checklist is good with exception about qsub falcon test. 
qsub falcon is currently under maintenance, and the qsub test under sawtooth is green. As CCB, I approve this PR and it can be merged into devel.

**Comment by wangcj05** [comment]
@joshua-cogliati-inl I have merged your PR, and thanks for your contribution.

**Pull request merged**